src/components/TableStateContainer.tsx

```diff
@@ -0,0 +1,351 @@
+                checked={queryCheckBoxState} 
+                onChange={()=>{
+                  if (queryCheckBoxState) setQueryCheckBoxState(false);
+                  else setQueryCheckBoxState(true)
+                }} 
+              />
+          </span>
+          :
+          null
+        }
+      </div>
+      
+      <div
+        style={{
+          visibility: checkBoxes.length > 0? "visible" : "hidden",
+          display: "flex",
+          justifyContent: "space-between",
+          backgroundColor: "var(--color-header)",
+          color: "var(--color-ligth-main-second)",
+        }}
+      >
+        <div 
+          style={{
+            paddingTop: "17px",
+            paddingBottom: "17px",
+            paddingLeft: "12px",
+          }}
+        >
+          {`${checkBoxes.length} items selected`}
+        </div>
+        <div>
+        {
+          checkBoxActions.map(checkboxAction=>{
+            return (
+              <button
+                onClick={()=>{
+                  // @ts-ignore
+                  const rows = tableData.filter((row) => {return getIfCheckBoxOfUuidIsSelected(row.uuid)})
+                  checkboxAction.actionFunction(rows, tableData, setTableData, ()=>fetchWithUrl(currentUrl), ()=>fetchWithUrl(url), setCheckBoxes)
+                }}
+                style={{
+                  border: "none",
+                  backgroundColor: "rgba(0,0,0,0)",
+                  color: "white",
+                  paddingTop: "17px",
+                  paddingBottom: "17px",
+                  paddingRight: "12px",
+                }}
+              >
+                {`${checkboxAction.displayValue} (${checkBoxes.length})`}
+              </button>
+            );
+          })
+        }
+        </div>
+      </div>
+      <div style={{flex:1, minHeight: 0}}>
+        <Table
+          tableData={dataWithCheckBoxes} 
+          setTableData={setTableData}
+          gridTemplateColumns={gridTemplateColumns} 
+          columnDefenitions={columnDefenitionsPlusCheckboxSortable}
+          dataRetrievalState={dataRetrievalState}
+          triggerReloadWithCurrentPage={()=>{fetchWithUrl(currentUrl)}}
+          triggerReloadWithBasePage={()=>{fetchWithUrl(url)}}
+          getIfCheckBoxOfUuidIsSelected={getIfCheckBoxOfUuidIsSelected}
+        />
+      </div>
+      
+      
+      <Pagination
+        page1Url={url}
+        previousUrl={previousUrl}
+        nextUrl={nextUrl}
+        itemsPerPage={itemsPerPage}
+        reloadFromUrl={fetchWithUrl}
+        setItemsPerPage={setItemsPerPage}
+      />
+
+    </div>
+  )
+};
+
+const mapStateToProps = (state:any, ownProps:any) => {
+  return {
+    bootstrap: state.bootstrap,
+    organisations: state.organisations
+  };
+};
+
+const mapDispatchToProps = (dispatch:any, ownProps:any) => {
+  return {
+  }
+};
+
+const TableStateContainer = withRouter(connect(mapStateToProps, mapDispatchToProps)(injectIntl(TableStateContainerElement)));
```

Comment:
We can remove `connect` if we remove `mapDispatchToProps` (not used now) and change `mapStateToProps` to `useSelector`. Not sure if withRouter is needed here. Do you know why we need it here?